lib/src/facts/linux/networking_resolver.cc

```diff
@@ -142,6 +142,17 @@ namespace facter { namespace facts { namespace linux {
         auto parse_route_line = [&known_route_types](string& line, std::vector<route>& routes) {
             vector<boost::iterator_range<string::iterator>> parts;
             boost::split(parts, line, boost::is_space(), boost::token_compress_on);
+
+            // remove trailing "onlink" or "pervasive" flags
+            for (;;) {
+                std::string last_token(parts.back().begin(),
```

Comment:
I don't think we need to make a copy here, since we don't reference it after we do the erasure. `auto& last_token = parts.back()` should be fine, and will be cleaner code to boot